app.js

```diff
@@ -1,18 +1,63 @@
-require('dotenv').config();
+require("dotenv").config();
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express");
+const hbs = require("hbs");
+const SpotifyWebApi = require("spotify-web-api-node");
 
 // require spotify-web-api-node package here:
+const spotifyApi = new SpotifyWebApi({
+  clientId: process.env.CLIENT_ID,
+  clientSecret: process.env.CLIENT_SECRET,
+});
+
+// Retrieve an access token
+spotifyApi
+  .clientCredentialsGrant()
+  .then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+  .catch((error) =>
+    console.log("Something went wrong when retrieving an access token", error)
+  );
 
 const app = express();
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs");
+app.set("views", __dirname + "/views");
+app.use(express.static(__dirname + "/public"));
+hbs.registerPartials(__dirname + "/views/partials");
 
 // setting the spotify-api goes here:
 
 // Our routes go here:
+app.get("/", (req, res) => {
+  res.render("homepage");
+});
+
+app.get("/artist-search", (req, res) => {
+  spotifyApi
+    .searchArtists(req.query.search)
+    .then((data) => {
+      console.log(
+        "The received data from the API: ",
+        data.body.artists.items[0]
+      );
+      res.render("artist-search", { artists: data.body.artists.items });
+    })
+    .catch((err) =>
+      console.log("The error while searching artists occurred: ", err)
+    );
+});
+
+app.get("/albums/:artistId", (req, res) => {
+  // .getArtistAlbums() code goes here
+  spotifyApi
+    .getArtistAlbums(req.params.artistId)
+    .then((data) => {
+      // console.log(data.body.items);
```

Comment:
also try to remove the console.logs everytime you create a PR its for good practice :P you can keep the ones inside catch